File: solve-boolean.cpp
#include <string>
using namespace std;

bool solveBoolean(string s) {
    bool result = true;
    for (int i = 0; i < s.length(); i++) {
        if (s[i] == 'T') {
            return true;
        }
        else if (s[i] == 'F') {
            return false;
        }
        else if (s[i] == '&') {
            result &= true;
        }
        else if (s[i] == '|') {
            result |= true;
        }
    }
    return result;
}

int main() {
    string s;
    cin >> s;
    cout << (solveBoolean(s) ? "True" : "False");
    return 0;
}

File: solve-boolean.py
def solve_boolean(expression):
    stack = []
    for char in expression[::-1]:
        if char == "T":
            stack.append(True)
        elif char == "F":
            stack.append(False)
        elif char == "&":
            b1 = stack.pop()
            b2 = stack.pop()
            stack.append(b1 and b2)
        elif char == "|":
            b1 = stack.pop()
            b2 = stack.pop()
            stack.append(b1 or b2)
    return stack[0]